makePDF.py

```diff
@@ -379,6 +383,12 @@ def run ( self ):
 
 			self.caller.output(content)
 			self.caller.output("\n\n[Done!]\n")
+
+			if _HAS_PHANTOMS:

```

Comment:
This block should be in the `try` block above. If the `parse_tex_log` line raises an exception, `errors`, `warnings` and `badboxes` will all be undefined raising another exception here and causing `finish()` to never be called. This also means that `errors`, `warnings` and `badboxes` probably need to be initialised in `run()` to empty lists.